Dockerfile

```diff
@@ -1,22 +1,32 @@
-FROM golang:1.11.4-alpine3.8 AS builder
-LABEL maintainer="joona@kuori.org"
+FROM golang:1.11.5-alpine3.8 AS builder
+
+ENV ACME_DNS_VERSION=v0.7
+
+WORKDIR /go/src/github.com/joohoi
 
-RUN apk add --update gcc musl-dev git
+RUN apk add --no-cache gcc musl-dev git && \
+    git clone -b ${ACME_DNS_VERSION} --depth 1 https://github.com/joohoi/acme-dns.git acme-dns
 
-RUN go get github.com/joohoi/acme-dns
 WORKDIR /go/src/github.com/joohoi/acme-dns
-RUN CGO_ENABLED=1 go build
 
-FROM alpine:latest
+RUN CGO_ENABLED=1 go build -ldflags="-s -w -linkmode external -extldflags -static"
 
-WORKDIR /root/
-COPY --from=builder /go/src/github.com/joohoi/acme-dns .
-RUN mkdir -p /etc/acme-dns
-RUN mkdir -p /var/lib/acme-dns
-RUN rm -rf ./config.cfg
-RUN apk --no-cache add ca-certificates && update-ca-certificates
+RUN mkdir -p /build/etc/acme-dns && \
+    mkdir -p /build/var/lib/acme-dns && \
+    cp /go/src/github.com/joohoi/acme-dns/acme-dns /build && \
+    cp /go/src/github.com/joohoi/acme-dns/config.cfg /build/etc/acme-dns && \
+    sed -Ei -e "s/(port ?= ?)\"80\"/\1\"8080\"/" \
+            -e "s/(listen ?= ?)\".*:53\"/\1\":5353\"/" /build/etc/acme-dns/config.cfg && \
```

Comment:
Can you think of a way to handle this without re-introducing #118 / #70 ?

Reply:
I don't know if this answers your question. But, as the container is running as a non root user, we cannot listen on a port < 1024. For that reason I added the `sed` lines here, to change the ports in the default config to ones that are higher than 1024. This "fix" can break, if the style of the config file changes and the regex does not match anymore. If you want I can add a test.sh file for that, that is executed on docker image build time. This script will then fail, if the port is not correct.
The second option will be, that we change the port in the default config file. Then I can remove my sed fix from the dockerfile.
What do you think?

Reply:
Yeah, let's keep it as it is. I don't know what I was thinking when I wrote the initial comment. The mentioned issues exists on the host side.

Reply:
Ok. Then I will resolve this conversation.